packages/ihe-gateway-sdk/src/models/patient-discovery/patient-discovery-responses.ts

```diff
@@ -0,0 +1,61 @@
+import * as z from "zod";
+import {
+  baseResponseSchema,
+  baseErrorResponseSchema,
+  operationOutcomeSchema,
+  externalGatewayPatientSchema,
+  XCPDGatewaySchema,
+} from "../shared";
+
+const patientDiscoveryRespSuccessfulDefaultSchema = baseResponseSchema.extend({
+  patientMatch: z.literal(true),
+  patientResource: z.object({}),
+  xcpdHomeCommunityId: z.string(),
+});
+
+// TO EXTERNAL GATEWAY
+const patientDiscoveryRespToExternalGWSuccessfulSchema =
+  patientDiscoveryRespSuccessfulDefaultSchema.extend({
+    patientMatchDegree: z.number().optional(),
+    externalGatewayPatient: externalGatewayPatientSchema,
+  });
+
+const patientDiscoveryRespToExternalGWFaultSchema = baseErrorResponseSchema.extend({
+  patientMatch: z.literal(false).or(z.literal(null)),
+  xcpdHomeCommunityId: z.string().optional(),
+});
+
+export const patientDiscoveryRespToExternalGWSchema = z.union([
+  patientDiscoveryRespToExternalGWSuccessfulSchema,
+  patientDiscoveryRespToExternalGWFaultSchema,
+]);
+
+export type PatientDiscoveryRespToExternalGW = z.infer<
+  typeof patientDiscoveryRespToExternalGWSchema
+>;
+
+// FROM EXTERNAL GATEWAY
+const patientDiscoveryRespFromExternalGWDefaultSchema = baseResponseSchema.extend({
+  gateway: XCPDGatewaySchema,
+  patientResourceId: z.string(),
+});
+
+const patientDiscoveryRespFromExternalGWSuccessfulSchema =
+  patientDiscoveryRespFromExternalGWDefaultSchema.merge(
+    patientDiscoveryRespSuccessfulDefaultSchema
+  );
+
+const patientDiscoveryRespFromExternalGWFaultSchema =
+  patientDiscoveryRespFromExternalGWDefaultSchema.extend({
+    patientMatch: z.literal(false),
```

Comment:
is this supposed to be null or false? On the patientDiscoveryRespToExternalGWFaultSchema shamil had said it should be null

Reply:
Got it that wasnt mentioned in Notion so i can update